producers/pairselection.py

```diff
@@ -1,427 +1,185 @@
-        nanoAOD.Electron_phi,
-        nanoAOD.Electron_mass,
-        q.good_electrons_mask,
-    ],
-    output=[q.dileptonpair],
-    scopes=["ee"],
-)
-ZElElPairSelection = Producer(
-    name="ZElElPairSelection",
-    call="ditau_pairselection::elel::ZBosonPairSelection({df}, {input_vec}, {output}, {pairselection_min_dR})",
-    input=[
-        q.Electron_pt_corrected,
-        nanoAOD.Electron_eta,
-        nanoAOD.Electron_phi,
-        nanoAOD.Electron_mass,
-        q.good_electrons_mask,
-    ],
-    output=[q.dileptonpair],
-    scopes=["ee"],
-)
-
-GoodElElPairFlag = Producer(
-    name="GoodElElPairFlag",
-    call="ditau_pairselection::flagGoodPairs({df}, {output}, {input})",
-    input=[q.dileptonpair],
-    output=[],
-    scopes=["ee"],
-)
-
-GoodElElPairFilter = Filter(
-    name="GoodElElPairFilter",
-    call='event::filter::Flags({df}, "GoodElElPairs", {input}, "any_of")',
-    input=[],
-    scopes=["ee"],
-    subproducers=[GoodElElPairFlag],
-)
-
-ETPairSelection = Producer(
-    name="ETPairSelection",
-    call="ditau_pairselection::eltau::PairSelection({df}, {input_vec}, {output}, {pairselection_min_dR})",
-    input=[
-        q.Tau_pt_corrected,
-        nanoAOD.Tau_eta,
-        nanoAOD.Tau_phi,
-        nanoAOD.Tau_mass,
-        nanoAOD.Tau_IDraw,
-        q.Electron_pt_corrected,
-        nanoAOD.Electron_eta,
-        nanoAOD.Electron_phi,
-        nanoAOD.Electron_mass,
-        nanoAOD.Electron_iso,
-        q.good_electrons_mask,
-        q.good_taus_mask,
-    ],
-    output=[q.dileptonpair],
-    scopes=["et"],
-)
-
-GoodETPairFlag = Producer(
-    name="GoodETPairFlag",
-    call="ditau_pairselection::flagGoodPairs({df}, {output}, {input})",
-    input=[q.dileptonpair],
-    output=[],
-    scopes=["et"],
-)
-
-GoodETPairFilter = Filter(
-    name="GoodETPairFilter",
-    call='event::filter::Flags({df}, "GoodElTauPairs", {input}, "any_of")',
-    input=[],
-    scopes=["et"],
-    subproducers=[GoodETPairFlag],
-)
+class kinematic_vars:
+    _nanoAOD_Muon = [nanoAOD.Muon_eta, nanoAOD.Muon_phi, nanoAOD.Muon_mass]
+    _nanoAOD_Tau = [nanoAOD.Tau_eta, nanoAOD.Tau_phi]
+    _nanoAOD_Electron = [nanoAOD.Electron_eta, nanoAOD.Electron_phi, nanoAOD.Electron_mass]
+
+
+kinematic_vars.Muon = [nanoAOD.Muon_pt, *kinematic_vars._nanoAOD_Muon]
+kinematic_vars.Tau = [nanoAOD.Tau_pt, *kinematic_vars._nanoAOD_Tau, nanoAOD.Tau_mass]
+kinematic_vars.Tau_with_corrected_pt = [q.Tau_pt_corrected, *kinematic_vars._nanoAOD_Tau, nanoAOD.Tau_mass]
+kinematic_vars.Tau_with_corrected_pt_and_mass = [q.Tau_pt_corrected, *kinematic_vars._nanoAOD_Tau, q.Tau_mass_corrected]
+kinematic_vars.Electron_with_correctrd_pt = [q.Electron_pt_corrected, *kinematic_vars._nanoAOD_Electron]
+
+with defaults(scopes=["mt"]):
+    MTPairSelection = Producer(
+        call="ditau_pairselection::mutau::PairSelection({df}, {input_vec}, {output}, {pairselection_min_dR})",
+        input=[
+            *kinematic_vars.Tau_with_corrected_pt,
```

Comment:
Here `Tau_with_corrected_pt_and_mass` should be used. Same for et and tt channel.